native/src/utils/PushNotificationsManager.ts

```diff
@@ -71,16 +77,47 @@ export const subscribeNews = async (city: string, language: string): Promise<voi
   }
 }
 
-const urlFromMessage = (message: Message): string => {
-  const { city_code: cityCode, language_code: languageCode, news_id: newsId } = (message as Message).data
-  return urlFromRouteInformation({
-    cityCode,
-    languageCode,
-    route: NEWS_ROUTE,
-    newsType: LOCAL_NEWS_TYPE,
-    newsId,
-  })
-}
+const routeInformationFromMessage = (message: Message): NonNullableRouteInformationType => ({
+  cityCode: message.data.city_code,
+  languageCode: message.data.language_code,
+  route: NEWS_ROUTE,
+  newsType: LOCAL_NEWS_TYPE,
+  newsId: message.data.news_id,
+})
+const urlFromMessage = (message: Message): string => urlFromRouteInformation(routeInformationFromMessage(message))
+
+export const useForegroundPushNotificationListener = ({
+  showSnackbar,
+  navigate,
+}: {
+  showSnackbar: (snackbar: SnackbarType) => void
+  navigate: (route: RoutesType, params: Record<string, unknown>) => void
+}): void =>
+  useEffect(() => {
+    let mounted = true
+    importFirebaseMessaging().then(messaging =>
+      messaging().onMessage(async _message => {
+        const message = _message as Message
+        if (mounted) {
+          // The CMS needs some time until the push notification is available in the API response
+          setTimeout(() => {
+            // TODO IGAPP-1024: Uncomment and improve snackbar
+            log(JSON.stringify(message))
+            // showSnackbar({
+            //   text: message.notification.title,
+            //   positiveAction: {
+            //     onPress: () => navigate(NEWS_ROUTE, routeInformationFromMessage(message)),
+            //     label: 'Show',
+            //   },
+            // })
+          }, WAITING_TIME_FOR_CMS)
```

Comment:
So this does not work without the setTimeout? I thought on message only triggers when there already is a message.

Reply:
There is already a message, yes. But its not delivered by the cms via the local news api. I will maybe ask them about that.

Reply:
So without the timeout the app just says "page not found" since there is no local news yet with that id if you click on the snackbar fast